run.sh

```diff
@@ -3,12 +3,16 @@
 set -x
 set -e
 
-ARGS=(--name vertx-fib-demo -p 8080:8080 -p 9093:9093)
+ARGS=(--name vertx-fib-demo -p $HTTP_PORT -p 9093:9093)
+
+if [ -z "$HTTP_PORT" ]; then
```

Comment:
I think this needs to be moved up before the `ARGS=` line - that one references `HTTP_PORT`, so the default value set should happen first.